GENERATORS/SIMULATIONS/BASIC_OSCILLATOR/BASIC_OSCILLATOR.py

```diff
@@ -46,7 +47,7 @@ def BASIC_OSCILLATOR(
         y = offset + amplitude * signal.square(2 * np.pi * frequency * x + phase)
     elif waveform == "triangle":
         y = offset + amplitude * signal.sawtooth(2 * np.pi * frequency * x + phase, 0.5)
-    else:  # Sawtooth
+    elif waveform == "sawtooth":
         y = offset + amplitude * signal.sawtooth(2 * np.pi * frequency * x + phase)
```

Comment:
Can you add an else and throw an exception there? Since in that case user passed in a incorrect input right?

Reply:
This is a select type, since there isn't a way for the user to input something else, I feel that is unnecessary

Reply:
Yes, but if you think about it, let's say we added a new select type, but forgot to update the if statement here. 
Without an explicit else statement, it is hard to catch bugs like this, since the code will not complain.

Reply:
It is just a good practice to have in general, in most cases, never hurt to do that, only adds benefit.

Reply:
Sounds good to me, but if we are going to do that it should be made clear to all node developers. Off the top of my head I don't think a lot of the nodes with select throws an exception.